include/fmt/core.h

```diff
@@ -422,11 +422,17 @@ template <typename Char> class basic_string_view {
    */
   FMT_CONSTEXPR_CHAR_TRAITS
   FMT_INLINE
-  basic_string_view(const Char* s) : data_(s) {
+  basic_string_view(const Char* s) : data_(s), size_(0) {
+    // This test is needed to ensure that this constructor is constexpr in C++17
+    // modes.
+#ifdef __cpp_lib_is_constant_evaluated
```

Comment:
``__cpp_lib_is_constant_evaluated`` is defined only in C++20. Therefore, code generation optimizations will not work until C++20.